src/Microsoft.Data.SqlClient/netcore/src/Common/src/System/Net/Logging/NetEventSource.Common.cs

```diff
@@ -518,6 +521,8 @@ private static string Format(FormattableString s)
         #region Custom WriteEvent overloads
 
         [NonEvent]
+        [SuppressMessage("ReflectionAnalysis", "IL2026:UnrecognizedReflectionPattern",
```

Comment:
Should it be `UnconditionalSuppressMessage`? It's available on .NET 5+ so it needs an `#if` guard.

Reply:
Yes - it MUST be `UnconditionalSuppressMessage`. `SuppressMessage` is "conditional" and the compiler will not emit it to IL - it disappears after compilation. As a result the trimmer doesn't get to see it.

We've been discussing some possibilities to provide an analyzer for these "bugs" but it's difficult (`SuppressMessage` is handled by the compiler intrinsically...). I could not find a tracking issue, so I created https://github.com/dotnet/runtime/issues/93453.